fbuapp/View Controllers/ProfileViewController.m

```diff
@@ -11,84 +11,225 @@
+        self.nameLabel.text = [NSString stringWithFormat:@"%@ %@", currentUser[@"firstName"], currentUser[@"lastName"]];
+        self.usernameLabel.text = [NSString stringWithFormat:@"@%@", currentUser.username];
+        self.bioLabel.text = currentUser[@"bio"];
+    }];
+    
+    //creating segment control bar
+    self.mainSegment = [[UISegmentedControl alloc] initWithItems:[NSArray arrayWithObjects:@"Posts", @"Saved", nil]];
+    self.mainSegment.frame = CGRectMake(87, 261, 200, 30);
+    self.mainSegment.layer.shadowOffset = CGSizeMake(0, 2.0f);
+    self.mainSegment.layer.shadowOpacity = 0.25f;
+    self.mainSegment.selectedSegmentIndex = 0;
+    [self.mainSegment addTarget:self action:@selector(fetchPosts) forControlEvents: UIControlEventValueChanged];
+    [self.profileView addSubview:self.mainSegment];
+    
     self.profilePictureView.layer.cornerRadius = self.profilePictureView.frame.size.width / 2;
     self.profilePictureView.clipsToBounds = YES;
-    //add border to profile image
     self.profilePictureView.layer.borderWidth = 3.0f;
     self.profilePictureView.layer.borderColor = [UIColor whiteColor].CGColor;
-    //Shadow and Radius of profile View
     self.profileView.layer.shadowColor = [[UIColor colorWithRed:0 green:0 blue:0 alpha:0.25f] CGColor];
     self.profileView.layer.shadowOffset = CGSizeMake(0, 2.0f);
     self.profileView.layer.shadowOpacity = 1.0f;
     self.profileView.layer.shadowRadius = 0.0f;
     self.profileView.layer.cornerRadius = 4.0f;
     
+    //initialize
+    self.postQuery = [Post query];
+    self.feed = [[PostTableView alloc] initWithUserId:[PFUser currentUser].objectId];
+    [self fetchPosts];
+    self.feed.delegate = self;
+    [self.postView addSubview:self.feed];
+    [self.feed mas_makeConstraints:^(MASConstraintMaker *make) {
+        make.edges.equalTo(self.postView).with.insets(UIEdgeInsetsMake(0, 0, 0, 0));
+    }];
+    
+    self.navigationItem.title = [NSString stringWithFormat:@"Profile"];
+}
+
+
+//load user's posts to their profile
+- (void)fetchPosts {
+    if(self.mainSegment.selectedSegmentIndex == 0)
+    {
+        [self checkCount];
+        ///query for posts that user created
+        [self.postQuery orderByDescending:@"createdAt"];
+        [self.postQuery whereKey:@"eventAuthor" equalTo:[PFUser currentUser].objectId];
+        [self.postQuery findObjectsInBackgroundWithBlock:^(NSArray<Post *> * _Nullable posts, NSError * _Nullable error) {
+            if (error) {
+                NSLog(@"%@", error.localizedDescription);
+            }
+            else {
+                NSLog(@"Loaded User's posts");
+                self.feed.posts = [NSArray arrayWithArray:posts];
+                NSLog(@"%@", posts);
+                [self.feed reloadData];
+            }
+        }];
+        [self.feed.refreshControl endRefreshing];
+    }
+    else if(self.mainSegment.selectedSegmentIndex == 1) {
```

Comment:
is it possible for self.mainSegment.selectedSegmentIndex to be something other than 0 or 1?